Added sector erase function

mickem opened the pull request:
This PR adds a sector erase function.

We use the library for testing with flash memory and programing takes to long as we do this frequently setting up test cases so we needed sector erase to speedup tests.
And while we could use chip erase that is slower thank sector erase for small areas.

This new function does not currently allow erasing partial sectors (which is checked) but this could be extended in the future with the same API but as I did not need it I did not add it to keep the complexity low as there are no tests.

If there is an interest for PR:s I could happily add a PR adding some tests as well as making some internal functions and the commands list public this as we use this command list to provide input in the lower SPI layers.

mickem committed:
Added function to erase sectors without writing new data

adamgreig commented:
Thanks for the PR! Looks good at a first glance but it's been ages since I wrote this so I'll need a little longer to double check it makes sense to me. I'd be happy to take other PRs too and can cut a new release whenever. Please could you remove the change to `.gitignore`? I prefer to keep editor specific things out of project ignore files; you can configure your git to globally ignore `.idea` if that's useful.

mickem commented:
Removed the gitignore file.

Also added some unit tests for the code which was touched by this change. As I did not find much test I have done "something" and opted to not bring in mocking libraries so please do let me know how you feel about this...
There are many ways to do testing in rust so not sure what you prefer...

mickem committed:
Added tests for code touched by sector_erase

mickem commented:
Any update on this? Maybe we should merge this and add a new version for before I submit othger PR:s?
